processor/cumulativetodeltaprocessor/tracking/identity.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracking
+
+import (
+	"bytes"
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+type MetricIdentity struct {
+	Resource               pdata.Resource
+	InstrumentationLibrary pdata.InstrumentationLibrary
+	MetricDataType         pdata.MetricDataType
+	MetricIsMonotonic      bool
+	MetricName             string
+	MetricUnit             string
+	StartTimestamp         pdata.Timestamp
+	Attributes             pdata.AttributeMap
```

Comment:
This and the `Write` implementation make me nervous because Go has built-in support for using structs as map keys. The OTel-Go SDK exports an `attribute.Set` type used for metric identity inside the SDK, see https://github.com/open-telemetry/opentelemetry-go/blob/92551d3933c9c7ef5eaf4f93f876a5487d0024b9/attribute/set.go#L198

With one of the `attribute.Distinct` types for the resource and one for the attributes, the only other point type here is `InstrumentationLibrary`. I'd be happy with a TODO to revisit this.